Partial rewrite of Google Scholar.js

zuphilip opened the pull request:
This PR is for discussion. @adam3smith has worked on GS separately, as he mentioned in https://github.com/zotero/translators/issues/1429#issuecomment-334933324 . Actually, I find it quite hard to read the GS translator code currently and some things might not be needed anymore. Thus, I decided to try to rewrite parts of GS translator.

This rewritten code works stable and fixes some of the problems mentioned in the forum or here. Moreover, it can handle author profile pages, which was also mentioned somewhere.

The attachments seem to work in Scaffold but not when I tried it outside. Some other smaller things should be further considered. And more testing of everything!

But maybe we don't need this things and can continue differently. That would be good to decide together.

adam3smith commented on Google Scholar.js:
```diff
@@ -9,61 +9,239 @@
+		} else if (type && type=="patent") {
+			//TODO
+		} else {
+			var related = ZU.xpathText(doc, '//a[contains(@href, "q=related:")]/@href');
```
See my much more stable solution in the new PR:
https://github.com/zotero/translators/pull/1437/files#diff-b47307f8aac17a6460d79fb3c632ee83R405

zuphilip replied:
Yes, I am also using `data-cid`. But these lines here are needed for a single page like https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u5HHmVD_uO8C or a profile page like https://scholar.google.de/citations?user=INQwsQkAAAAJ&hl=de&oi=sra .

adam3smith replied:
I see -- but wouldn't it be more reliable to use/parse the title link for this, i.e.
href="http://books.google.com/books?hl=en&lr=&id=tP8HLETgbKcC&oi=fnd&pg=PA205&dq=info:54GDwtCuzVwJ:scholar.google.com&ots=-hJyQuF2DD&sig=xPUWsIsTJzFHmglCmQCHo4ARcFU"
You already look this up at `attr(context, 'h3 a, #gsc_vcd_title a', 'href');`
the itemID is in there, too, and that's presumably guaranteed to exist for every item, which isn't the case for related.

zuphilip replied:
Here is another example https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u-x6o8ySG0sC where the extraction from the related works, but the title link is simply http://www.springerlink.com/index/rm32474088w54378.pdf

adam3smith replied:
OK. Would it make sense to check for both to catch even more cases?

zuphilip replied:
AFAS this url parameter is only present in links to GoogleBooks.

adam3smith replied:
OK, agree that's probably not worth it then.

adam3smith commented:
I'm otherwise happy for you to run with this. It's working well for me and I do get PDFs after restarting Zotero and Firefox. Let me know when you're ready for final review.

zuphilip commented:
@adam3smith Did you test it in Firefox?

adam3smith commented:
Yes, FF56, 64-bit on linux with the 5.0.24 connector

zuphilip commented on Google Scholar.js:
```diff
@@ -9,61 +9,239 @@
-		} else if(link.indexOf('/scholar_case?') != -1) {
+		
+		if (link.indexOf('/patents?') != -1) {
+			return 'patent';// TODO no example found
```
I think that there are no examples for single page about patent in GoogleScholar and because they are now all on Google directly, e.g. https://www.google.com/patents/US20160328455, I am quite certain we don't need this case anymore. If a patent occurs among the search results in Google Scholar, then we will still be able to handle that.

zuphilip commented:
Okay, I now rewrote parts of the translator and fixed thereby several issues. I didn't touched the part about cases from @fbennett, but it would be good if you can test it as well. It works well for me i) for search pages (also for the [recently reported issues](https://forums.zotero.org/discussion/comment/288027#Comment_288027)), ii) profile pages #1376 CC @mvolz , iii) individual page for articles, iv) individual page for cases. It saves a snapshot and pdf #1429 CC @dstillman if possible.

One problem, I currently see, is that selecting multiple items on profile pages does not work for me in Firefox (but it works in Chrome). @adam3smith Is this reproducible?

adam3smith commented:
Great, thanks -- no complaints about the code & all tests and examples work for me, including profile pages on Firefox.
